src/EditorFeatures/CSharpTest/Diagnostics/AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest.cs

```diff
@@ -138,5 +139,12 @@ public interface IAsyncEnumerator<out T> : IAsyncDisposable
         T Current { get; }
     }
 }";
+
+        internal IOptionsCollection RequireArithmeticBinaryParenthesesForClarity => ParenthesesOptionsProvider.RequireArithmeticBinaryParenthesesForClarity;
+        internal IOptionsCollection RequireRelationalBinaryParenthesesForClarity => ParenthesesOptionsProvider.RequireRelationalBinaryParenthesesForClarity;
+        internal IOptionsCollection RequireOtherBinaryParenthesesForClarity => ParenthesesOptionsProvider.RequireOtherBinaryParenthesesForClarity;
+        internal IOptionsCollection IgnoreAllParentheses => ParenthesesOptionsProvider.IgnoreAllParentheses;
+        internal IOptionsCollection RemoveAllUnnecessaryParentheses => ParenthesesOptionsProvider.RemoveAllUnnecessaryParentheses;
+        internal IOptionsCollection RequireAllParenthesesForClarity => ParenthesesOptionsProvider.RequireAllParenthesesForClarity;
```

Comment:
can you just move these up to the base type?  that way you don't need to repeat for vb, rigt?

Reply:
The base type is actually in an assembly that does not have IVT to CodeStyle layer, because it needs IVT to Workspaces layer to execute tests using IDE test framework. We can potentially clean this up by introducing another intermediate test utility assembly, but I felt that would be an overkill, especially given we are planning to port these tests to Microsoft.CodeAnalysis.Testing, at which point these files can be deleted.